nilearn/tests/test_masking.py

```diff
@@ -1,371 +1,483 @@
-    with pytest.raises(DimensionError, match=_TEST_DIM_ERROR_MSG % "4D"):
-        masking.apply_mask(data_img, mask_img_4d)
 
-    # Check that 3D data is accepted
-    data_3d = Nifti1Image(np.arange(27, dtype="int32").reshape((3, 3, 3)),
-                          np.eye(4))
-    mask_data_3d = np.zeros((3, 3, 3))
+def test_apply_mask_3D_data_is_accepted():
+    shape = (3, 3, 3)
+    data_3d = Nifti1Image(
+        np.arange(27, dtype="int32").reshape(shape), AFFINE_EYE
+    )
+
+    mask_data_3d = np.zeros(shape)
     mask_data_3d[1, 1, 0] = True
     mask_data_3d[0, 1, 0] = True
     mask_data_3d[0, 1, 1] = True
-    data_3d = masking.apply_mask(data_3d, Nifti1Image(mask_data_3d, np.eye(4)))
-    assert sorted(data_3d.tolist()) == [3., 4., 12.]
+
+    data_3d = apply_mask(data_3d, Nifti1Image(mask_data_3d, AFFINE_EYE))
+
+    assert sorted(data_3d.tolist()) == [3.0, 4.0, 12.0]
+
+
+def test_apply_mask_errors():
+    data = np.zeros(SHAPE_4D)
+    data_img = Nifti1Image(data, AFFINE_EYE)
+
+    # veriy that 4D masks are rejected
+    mask_img_4d = Nifti1Image(np.ones(SHAPE_4D), AFFINE_EYE)
+    with pytest.raises(DimensionError, match=_TEST_DIM_ERROR_MSG % "4D"):
+        apply_mask(data_img, mask_img_4d)
 
     # Check data shape and affine
+    mask = np.ones(SHAPE_3D)
     with pytest.raises(DimensionError, match=_TEST_DIM_ERROR_MSG % "2D"):
-        masking.apply_mask(data_img,
-                           Nifti1Image(mask[20, ...], affine))
-    pytest.raises(ValueError, masking.apply_mask,
-                  data_img, Nifti1Image(mask, affine / 2.))
+        apply_mask(data_img, Nifti1Image(mask[5, ...], AFFINE_EYE))
+
+    with pytest.raises(ValueError, match="Mask affine"):
+        apply_mask(
+            data_img,
+            Nifti1Image(mask, AFFINE_EYE / 2.0),
+        )
+
     # Check that full masking raises error
-    pytest.raises(ValueError, masking.apply_mask,
-                  data_img, full_mask_img)
+    full_mask = np.zeros(SHAPE_3D)
+    full_mask_img = Nifti1Image(full_mask, AFFINE_EYE)
+    with pytest.raises(ValueError, match="The mask is invalid as it is empty"):
+        apply_mask(data_img, full_mask_img)
+
     # Check weird values in data
-    mask[10, 10, 10] = 2
-    pytest.raises(ValueError, masking.apply_mask,
-                  data_img, Nifti1Image(mask, affine))
-    mask[15, 15, 15] = 3
-    pytest.raises(ValueError, masking.apply_mask,
-                  Nifti1Image(data, affine), mask_img)
+    mask[5, 5, 5] = 2
+    with pytest.raises(
+        ValueError, match="Background of the mask must be represented with 0"
+    ):
+        apply_mask(data_img, Nifti1Image(mask, AFFINE_EYE))
+
+    mask[6, 6, 6] = 3
+    mask_img = Nifti1Image(mask, AFFINE_EYE)
+    with pytest.raises(ValueError, match="Given mask is not made of 2 values"):
+        apply_mask(Nifti1Image(data, AFFINE_EYE), mask_img)
 
 
-def test_unmask():
-    # A delta in 3D
-    shape = (10, 20, 30, 40)
+def test_unmask_4D():
```

Comment:
```suggestion
def test_unmask_4d():
```
@NicolasGensollen missed one ;)